ur-packages/hex-string/__tests__/hexstring.test.ts

```diff
@@ -0,0 +1,104 @@
+import { HexString } from "../src";
+import { URRegistryDecoder } from "@keystonehq/bc-ur-registry";
+
+describe("hex-string", () => {
+    const hex = "babecafe8badf00d";
+    const buff = Buffer.from(hex, "hex");
+
+    const expectedCBOR = "D9010748BABECAFE8BADF00D".toLowerCase(); // Should we?
+    const expectedUR = "ur:hex-string/taadatfdrdrnsgzelupmwtbtbtdsvlse";
+
+    it("should generate hex-string from buffer", () => {
+
+        // Create hex-string from buffer
+        const hexString = new HexString(buff);
+
+        // Convert it to cbor then hex
+        const cborHex = hexString.toCBOR().toString("hex");
+        /**
+         * CBOR should contain
+         * 263(h'babecafe8badf00d')
+         *
+         * ```
+         * D9 0107                # tag(263)
+         *    48                  # bytes(8)
+         *     BABECAFE8BADF00D # "\xBA\xBE\xCA\xFE+\xAD\xF0\r"
+         * ```  
+         */
+
+        expect(cborHex).toBe(expectedCBOR);
+
+        const ur = hexString.toUREncoder(1000).nextPart();
+
+        expect(ur).toBe(expectedUR);
+    });
+
+    it("should generate hex-string from hex string", () => {
+
+        // Create hex-string from buffer
+        const hexString = new HexString(hex);
+
+        const cborHex = hexString.toCBOR().toString("hex");
+        expect(cborHex).toBe(expectedCBOR);
+
+        const ur = hexString.toUREncoder(1000).nextPart();
+        expect(ur).toBe(expectedUR);
+    });
+
+    it("should decode hex-string from cbor", () => {
+        const hexString = HexString.fromCBOR(Buffer.from(expectedCBOR, "hex"));
+        expect(hexString.toHex()).toBe(hex);
+        
+    });
+
+    it("should decode hex-string from ur", () => {
+        const decoded = URRegistryDecoder.decode(expectedUR);
+
+        const hexString = HexString.fromCBOR(decoded.cbor);
+
+        expect(hexString.getRegistryType().getType()).toBe("hex-string");
+        expect(hexString.toHex()).toBe(hex);
+        expect(hexString.toCBOR().toString("hex")).toBe(expectedCBOR);
+    });    
+
+
+    it("should encode and decode same buffer after QR flow", () => {
+        // Create hex-string from buffer
+        const hexString = new HexString(buff);
+
+        // Convert it to cbor then hex
+        const cborHex = hexString.toCBOR().toString("hex");
```

Comment:
so `hexString.toHex() !== hexString.toCBOR().toString("hex")` (cborHex)?

Reply:
Nope our data is: `BABECAFE8BADF00D`
CBOR data is: 
```
D9 0107           # tag(263)
   48                  # bytes(8)
      BABECAFE8BADF00D # "\xBA\xBE\xCA\xFE\x8B\xAD\xF0\r"
```